model/database/registry.yaml

```diff
@@ -541,9 +542,50 @@ groups:
               value: "ConsistentPrefix"
               stability: experimental
         stability: experimental
-        brief: Account or request [consistency level](https://learn.microsoft.com/azure/cosmos-db/consistency-levels).
+        brief: >
+          Effective Request [consistency level](https://learn.microsoft.com/azure/cosmos-db/consistency-levels).
+          This represents the first applicable consistency level among those specified by the request or the client.
         examples: ["Eventual", "ConsistentPrefix", "BoundedStaleness", "Strong", "Session"]
 
+  - id: registry.db.cosmosdb.network
+    type: attribute_group
+    display_name: Azure Cosmos DB Network level Attributes
+    stability: experimental
+    brief: >
+      This group defines attributes for Azure Cosmos DB Network call.
+    attributes:
+      - id: db.cosmosdb.network.host
+        type: string
+        stability: experimental
+        brief: Specifies the host address from the Gateway or Backend Service endpoint.
+        examples: ["rntbd://cdb-east-us.azure.document.com"]
+      - id: db.cosmosdb.network.port
+        type: int
+        stability: experimental
+        brief: The port used for the Gateway or Backend Service endpoint.
+        examples: 443
+      - id: db.cosmosdb.network.response.status_code
```

Comment:
Do we need to introduce new attribute for this? And should it be in `db.cosmosdb.network` ?

If it belongs in DB domain, let's reuse `db.response.status_code` instead.
If it's a protocol-specific thing, let's move it to the `rntbd.*` namespace

Reply:
It is DB network call specific, which can be `http` or `rntbd` call. Basically, for an operation, SDK makes multiple network calls (`rntbd` for Direct Mode , `http` for Gateway Mode)

Reply:
why we can't use `db.response.status_code` for it?

Reply:
its already there as a dimension to represent operation level statuscode.

Reply:
same comment as in https://github.com/open-telemetry/semantic-conventions/pull/1543#discussion_r1833661536 - I don't understand how you can report overall operation status on a specific network call. 

It'd mean that you need to accumulate all telemetry and report it only when overall operation ends. It'd lead to all sorts of implementation issues and edge cases, would increase memory consumption, and would keep metrics less real-time (potentially minutes). 

In case there is a huge amount of timing out operations, you'd have a hard time with providing observability into that's going on. 

In case of multi-request operations it becomes even more fun since you'd have to wait for all of them to complete and the status of the overall operation you report on a specific network request will have nothing to do with this request.

Reply:
SDK is already designed to have this information available after the operation is completed. you can refer this PR https://github.com/Azure/azure-cosmos-dotnet-v3/pull/4872

Reply:
Maybe to step back and provide some context. For debuggability in the Cosmos SDKs with direct mode we already carry a state bag CosmosDiagnostics through all layers and add information needed for debuggability. Each operation/API call allows getting the CosmosDiagnostics from CosmosException or the Response in success cases. We provide an option to specify diagnostics thresholds to control when to emit additional debug info (whether to include the string/json representation of CosmosDiagnostics in logs/traces, whether to emit network lever metrics etc.). Why is this needed? We promise customers in the COsmos DB SLA not only availability SLAs from the end-client's perspective (vs. most other Azure services only looking at server-side availability) but the same for latency under certain conditions (10ms end-to-end latency). This means customers sometimes come and expect RCA/explanation why a (small) subset of operations misses SLA. And collectiong the Comos Diagnostics info in logs/traces network metrics all the time would be way to noisy and expensive/inefficient. And the decision whether diagnostics are needed can only be made on the operation level (an operation could very well have one slow network call - but due to various retry policies succeed successfully quickly - at that point we would not want to consider it a violation of diagnostics thresholds. So, we will have to emit logs, network metrics and even traces after the fact for network calls based on the CosmosDiagnostics collected anyway. Java is already doing this - it works well and also helps encapsulating the different diagnostics processing logic. So, I think Sourabh's approach for .Net makes sense as well.